cnf-certification-test/lifecycle/suite.go

```diff
@@ -484,3 +489,29 @@ func testPodPersistentVolumeReclaimPolicy(env *provider.TestEnvironment) {
 		ginkgo.Fail(errMsg)
 	}
 }
+
+func testCPUIsolation(env *provider.TestEnvironment) {
+	ginkgo.By("Testing pods for CPU isolation requirements")
+
+	// Individual requirements we are looking for:
+	//  - CPU Requests and Limits must be in the form of whole units
+	// - Resource Requests and Limits must be provided and identical
+
+	// Additional checks if the above pass
+	// - 'runtimeClassName' must be specified
+	// - Annotations must be provided disabling CPU and IRQ load-balancing.
+
+	podsMissingIsolationRequirements := make(map[string]bool)
+
+	for _, put := range env.Pods {
+		if put.IsPodGuaranteed() && !put.IsCPUIsolationCompliant() {
```

Comment:
optional: split this if and count the # of guaranteed pods so that if it is zero, the test can be skipped instead of passing (e.g. no isolated pods to test). Something like: 
var guaranteedPods int
if put.IsPodGuaranteed() {
guaranteedPods++
if !put.IsCPUIsolationCompliant() {

Reply:
Either way, it'll be evaluating the `IsPodGuaranteed` for each pod correct?  I think I like this better than counters?  What do you think?

Reply:
I will be adding a `GuaranteedPods` variable to the provider env to be able to loop through (or skip if empty) the GuranteedPods.  👍

Reply:
Completed in: https://github.com/test-network-function/cnf-certification-test/commit/053908124e1c2a9065bec8f165ed27c99669a13c